File: Easy/100-Same Tree/100.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        
        if(!p || !q)
            return p == q;    
        else if(p->val != q->val)
            return false;
        
        bool left_equal = isSameTree(p->left, q->left);
        bool right_equal = isSameTree(p->right, q->right);
        
        return left_equal && right_equal;
    }
};

File: Easy/100-Same Tree/100.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        if not p or not q:
            return p == q
        elif p.val != q.val:
            return False
        
        left_same  = self.isSameTree(p.left, q.left)
        right_same = self.isSameTree(p.right, q.right)
        
        return left_same and right_same
        
        